### PR TITLE
`Programming exercises`: Add link to open external repository

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Steps for Testing

Prerequisites:
- 1 Instructor
- TS *without* LocalVC

1. Log in to Artemis
2. Navigate to Programming Exercise Management Detail Page
3. Verify that there is en "Open Repository" button for all the repositories, which opens the external repository.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **New Features**
	- Added conditional rendering for repository links in the detail overview list, enhancing navigation.
	- Introduced a new icon for external links, improving visual cues for users.

- **Enhancements**
	- Updated the display logic for repository buttons and links in programming exercise details for better user experience.

### CODE REVIEW

**Review by konrad2002 — APPROVED** [review]
manually tested on ts4, works as describe

**Review by MichaelOwenDyer — DISMISSED** [review]
Tested on TS4, everything works :)

**Review by farisd16 — DISMISSED** [review]
tested on ts4, works as expected

**Review by LenaKahle — DISMISSED** [review]
Approve, tested on ts4.

**Review by MaximilianJG — DISMISSED** [review]
Tested on ts4. Works as expected
Screenshot 2024-01-18 at 17 36 22

**Review by kaancayli — APPROVED** [review]
Approved. Tested on ts4

**Review by xHadie — APPROVED** [review]
_Manually tested on TS2_Legacy_
Works as described

**Review by coolchock — DISMISSED** [review]
code lgtm

**Review by mateusmm01 — DISMISSED** [review]
Code lgtm 👍

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
             "templateBuildPlanId": "Vorlagen-Build-Plan-Id",
             "solutionBuildPlanId": "Lösungs-Build-Plan-Id",
             "repositories": "Repositories",
+            "openRepositories": "Repository Öffnen",
```
Comment (rabeatwork) [inline src/main/webapp/i18n/de/programmingExercise.json:100]: I guess it should be "openRepository" instead of "openRepositories"

**Review by rabeatwork — APPROVED** [review]

**Review by konrad2002 — APPROVED** [review]

**Review by kaancayli — APPROVED** [review]
Reapprove

**Review by basak-akan — APPROVED** [review]
code lgtm

**Review by MaximilianJG — APPROVED** [review]

**Review by farisd16 — APPROVED** [review]

**Pull request merged**